File: javascript/leetcode/204. Count Primes.js
/**
 * @param {number} n
 * @return {number}
 */
var countPrimes = function (n) {
  let count = 0;
  const nums = new Array(n);
  nums.fill(true);

  for (let i = 2; i * i < n; ++i) {
    for (let j = 2; j * i < n; ++j) nums[j * i] = false;
  }

  for (let i = 2; i < n; ++i) {
    if (nums[i]) ++count;
  }

  return count;
};

File: c++/leetcode/204. Count Primes.cpp
class Solution
{
public:
  int countPrimes(int n)
  {
    vector<bool> nums(n, true);
    int count = 0;

    for (int i = 2; i * i < n; ++i)
    {
      for (int j = 2; j * i < n; ++j)
        nums[j * i] = false;
    }

    for (int i = 2; i < n; ++i)
    {
      if (nums[i])
        ++count;
    }

    return count;
  }
};